compiler/rustc_monomorphize/src/errors.rs

```diff
@@ -70,37 +70,23 @@ pub(crate) struct UnknownCguCollectionMode<'a> {
 }
 
 #[derive(LintDiagnostic)]
-#[diag(monomorphize_abi_error_disabled_vector_type_def)]
+#[diag(monomorphize_abi_error_disabled_vector_type)]
 #[help]
-pub(crate) struct AbiErrorDisabledVectorTypeDef<'a> {
+pub(crate) struct AbiErrorDisabledVectorType<'a> {
     #[label]
     pub span: Span,
     pub required_feature: &'a str,
     pub ty: Ty<'a>,
+    /// Whether this is a problem at a call site or at a declaration.
+    pub call: bool,
```

Comment:
`is_call` here too. (Or a new binary enum would be even better, though I can see that might not pass through easily to the switch in the `.ftl` file.)

Reply:
A new enum would have to be wired up with the diagnostics infrastructure and AFAIK we don't have a `derive` for that. That's why I picked a `bool`.

Will rename to `is_call`.